docs/datastorage/capabilities.md

```diff
@@ -0,0 +1,176 @@
+}
+```
+
+Attaching Capabilities
+----------------------
+
+As mentioned, attaching capabilities to entities and itemstacks can be done through the AttachCapabilityEvent. This event has 3 sub-events, for extra granularity:
+* `AttachCapabilityEvent.Entity`: Fires only for entities.
+* `AttachCapabilityEvent.TileEntity`: Fires only for tile entities.
+* `AttachCapabilityEvent.Item`: Fires only for item stacks.
+
+In all cases, the event has a method `addCapability`, which can be used to attach capabilities to the target object. Instead of adding capabilities themselves to the list, you add capability providers, which have the chance to return capabilities only from certain sides. While the provider only needs to implement `ICapabilityProvider`, it is good idea to implement `ICapabilitySerializable<T extends NBTBase>`, which on top of returning the capabilities, will allow providing NBT save/load functions.
+
+For information on how to implement `ICapabilityProvider`, refer to the [Exposing a Capability](#exposing-a-capability) section.
+
+Creating Your Own Capability
+----------------------------
+
+In general terms, a capability is declared and registered through a single method call.
+
+First of all, we will create our feature as a skeleton class or interface. For the purpose of adding data to an entity (replacing an IEEP), the feature can be just a class, but for an API, you will most probably want to design it around an interface.
+
+Although the capability instance itself will be initialized internally by Forge, we'll start by creating a class that manages the capability features. In this class, we will create a field to hold the instance of the capability.
+
+```Java
+@CapabilityInject(IExampleCapability.class)
+public static Capability<IExampleCapability> INSTANCE = null;
+```
+
+Then, we will be defining a static `register()` method, which will call `CapabilityManager.INSTANCE.register()`. For the purpose of this documentation we will be describing each part as a separate named class, although anonymous classes are an option.
+
+```Java
+public static void register()
+{
+  CapabilityManager.INSTANCE.register(IExampleCapability.class, new Storage(), new Factory());
+}
+```
+
+The first parameter to this method, is the type that describes the capability feature. In our example, this type refers to an interface.
+
+The second parameter is an instance of a class that implements `Capability.IStorage<T>`, where T is the same class we specified in the first parameter. This storage class will help manage saving and loading for the default implementation, and it can, optionally, also support other implementations.
+
+For the purpose of our example, we will make the capability interface implement the `INBTSerializable<NBTTagCompound>` interface, which will allow us to delegate the saving and loading to the instance itself, but custom saving and loading are possible, and depending on the case, maybe even preferred.
```

Comment:
It's entirely preferred, the entire point is to not require the interface to do it :/